### PR TITLE
Support pseudodata in combinetf

### PR DESCRIPTION
**Pull request by davidwalter2**
Add support of pseudodata fits. The pseudodata sets are filled as n x m matrix with n the number of bins and m the index for the pseudodata set. The pseudo data set of e.g. index 1 is fit via:
```
combinetf.py -p 1 card.hdf5
```
Additionally, the histogram name, axis name, and bin index from which the pseudodata is generated are loaded from "hpseudodatasystidxs" and can be used to specify the pseudodata set by string e.g. as
```
combinetf.py -p massWeightZ_massShift_massShiftZ20MeVDown card.hdf5
```

### CODE REVIEW

**Comment by kdlong** [comment]
We had discussed this on the Mattermost with @cippy. As I understand it, you can currently swap the data hist during text2hdf5. But having this feature enabled so that we can use the direct hdf5 output for the pseudo data scans is very useful

**Comment by cippy** [comment]
> We had discussed this on the Mattermost with @cippy. As I understand it, you can currently swap the data hist during text2hdf5. But having this feature enabled so that we can use the direct hdf5 output for the pseudo data scans is very useful

If needed, the option I had in mind about this topic is here [0], which is then inherited by text2hdf5. 
The idea is that in the txt datacard one can add a line to define a new histogram myPseudodata (specifying the file where to get it and how the object is named inside the file, as usual), and then one could simply use it in place of data_obs through -D/--dataset myPseudodata (whose default is data_obs).

[0] https://github.com/bendavid/HiggsAnalysis-CombinedLimit/blob/tensorflowfit/python/DatacardParser.py#L15

File: scripts/combinetf.py [hunk 1]
```diff
@@ -2034,8 +2060,10 @@ def fillHists(tag, witherrors=options.computeHistErrors):
   tndof[0] = x.shape[0]
   tndofpartial[0] = npoi
   ttaureg[0] = taureg
-  
-  for output,outputname, outvals,outsigmas,minosups,minosdowns,outgenvals,toutvals,touterrs,toutminosups,toutminosdowns,toutgenvals in zip(outputs, outputnames, outvalss,outsigmass,outminosupss,outminosdownss,outvalsgens,toutvalss,touterrss,toutminosupss,toutminosdownss,toutgenvalss):
+
+  for output,outputname, outvals,outsigmas,minosups,minosdowns,outgenvals,toutvals,touterrs,toutminosups,toutminosdowns,toutgenvals in zip(
+    outputs, outputnames, outvalss,outsigmass,outminosupss,outminosdownss,outvalsgens,toutvalss,touterrss,toutminosupss,toutminosdownss,toutgenvalss
+    ):
```
Comment (bendavid) [inline scripts/combinetf.py:2057]: is this a real change, or just whitespace?

Comment (davidwalter2) [reply scripts/combinetf.py:2057]: Just a whitespace/linebreak

**Comment by bendavid** [comment]
Can you explain a bit better how this simplified version works?  What is the indexing scheme exactly?  And is the --pseudodata argument really a string, or rather an index?

**Comment by davidwalter2** [comment]
Yes the --pseudodata argument is a string that is matched to the pseudo dataset names that are stored in "hpseudodatasystidxs". It seems safer to me to make a string comparison instead of picking the pseudodata based on an index where one would have to keep track of the index by oneself.

**Comment by davidwalter2** [comment]
I have now also renamed the 'hpseudodatasystidxs' collection into 'hpseudodatanames' which probably describes better what is stored there

**Pull request merged**